### PR TITLE
docs: added 'Data storage locations in zot' to Developer Guide

### PR DESCRIPTION
**Pull request by mbshields**
**What type of PR is this?**

documentation

**Which issue does this PR fix**:

none

**What does this PR do / Why do we need it**:

Provides a general explanation of where and how repo data and process data is stored in zot memory, including database names and paths.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: docs/developer-guide/onboarding.md [hunk 1]
```diff
@@ -130,3 +130,62 @@ The zot project codebase is organized as follows:
...
+		<th>Description</th>
+	</tr>
+	<tr>
+		<td>images</td>
+		<td>local and<br/>AWS S3</td>
+		<td>Image blobs</td>
+	</tr>
+	<tr>
+		<td>repository synchronization</td>
+		<td>local</td>
```
[the remark this answers is not in the document]
Comment (peusebiu) [reply docs/developer-guide/onboarding.md:151]: let's say zot syncs alpine repo:
1) it copies image blobs from upstream to ${zot_root_dir}/alpine/.sync/${UUID}/alpine/ temp dir.
2) it copies image blobs from ${zot_root_dir}/alpine/.sync/${UUID}/alpine/ temp dir to ${zot_root_dir}/alpine/ (sync is complete)
3) removes temp dir ${zot_root_dir}/alpine/.sync/${UUID}/

[the remark this answers is not in the document]
Comment (mbshields) [reply docs/developer-guide/onboarding.md:151]: Explanded explanation to:

| Data or Process | Storage Location |  Description |
| ---------------- | ----------------- | ------------ |
| repository synchronization | local /<repo_name>/.sync | The sync operation temporarily copies the upstream blobs to /<repo_name>/.sync/${UUID}/<repo_name>, then copies to /<repo_name> and deletes the temporary directory ${UUID} |

File: docs/developer-guide/onboarding.md [hunk 2]
```diff
@@ -130,3 +130,62 @@ The zot project codebase is organized as follows:
     - examples/         # Configuration examples
     - swagger/          # Swagger integration
     - docs/             # Documentation
+
+## Data storage locations in zot
```
[the remark this answers is not in the document]
Comment (mbshields) [reply docs/developer-guide/onboarding.md:134]: Changed section title

**Review by andaaron — APPROVED** [review]

**Review by rchincha — APPROVED** [review]
lgtm

**Pull request merged**